circuits/aes-gcm/utils.circom

```diff
@@ -403,15 +382,20 @@ template Contains(n) {
     out <== 1 - someEqual.out;
 }
 
+/// m is the number of arrarys, n is the length of each array
 template ArraySelector(m, n) {
     signal input in[m][n];
-    signal input index;
+    signal input index; 
     signal output out[n];
     assert(index >= 0 && index < m);
 
     signal selector[m];
+    component Equal[m];
     for (var i = 0; i < m; i++) {
-        selector[i] <-- index == i ? 1 : 0;
+        Equal[i] = IsEqual();
```

Comment:
if it helps, we can do this is a single line using [anonymous components](https://docs.circom.io/circom-language/anonymous-components-and-tuples/#the-use-of-for-signal-arrays).

```
selector[i] <== IsEqual()([index, i]);
```

Reply:
This is cool! Made the switch, it's for sure a little cleaner